core/tests/test_adding_post.py

```diff
@@ -0,0 +1,68 @@
+import secrets
+import string
+
+import pytest
+from django.contrib.auth import get_user_model
+from django.test import Client
+from django.urls import reverse
+
+from core.models import Community
+
+pytestmark = pytest.mark.django_db
+
+User = get_user_model()
+
+
+def generate_random_password(length: int = 12) -> str:
+    characters = string.ascii_letters + string.digits + string.punctuation
+    return "".join(secrets.choice(characters) for _ in range(length))
+
+@pytest.fixture()
+def user(client: Client) -> User:
+    password = generate_random_password()
+    user = User.objects.create_user(
+        email="test_user@example.com",
+        nickname="TestUser",
+        password=password,
+    )
+
+    client.login(email=user.email, password=user.password)
+    return user
+
+@pytest.fixture()
+def community() -> Community:
+    return Community.objects.create(name="Test Community", is_active=True)
+
+
+def test_add_post_valid(client: Client, user: User, community: Community) -> None:
+    data = {
+        "community": community.pk,
+        "title": "Test Post Title",
+        "content": "This is a test post content.",
+    }
+    client.force_login(user)
+    response = client.post(reverse("add-post"), data=data, follow=True)
+    assert response.status_code == 200
+    assert response.context["post"].author == user
+    assert response.context["post"].title == data["title"]
+    assert response.context["post"].content == data["content"]
+
+
+def test_add_post_invalid(client: Client, user: User, community: Community) -> None:
+    data = {"title": ""}  # Missing content
+    client.force_login(user)
+    response = client.post(reverse("add-post"), data=data)
+    assert response.status_code == 200
+    assert b"This field is required." in response.content
```

Comment:
This could be slightly improved. It shows field is required, but we don't know which field. 
One way to do it would be, probably, to do something like. 
response.context.form.fields["title'].errors.0 == ...

Reply:
done